src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -42,4 +45,21 @@ class ReferralController(private val repository: ReferralRepository) {
       ?.let { ResponseEntity.ok(DraftReferral(it)) }
       ?: ResponseEntity.notFound().build()
   }
+
+  @PatchMapping("/draft-referral/{id}")
+  fun patchDraftReferralByID(@PathVariable id: String, @RequestBody partialUpdate: DraftReferral): ResponseEntity<Any> {
+    val uuid = try {
+      UUID.fromString(id)
+    } catch (e: IllegalArgumentException) {
+      return ResponseEntity.badRequest().body("malformed id")
+    }
+
+    return repository.findByIdOrNull(uuid)
+      ?.let {
+        ReferralService.updateReferral(it, partialUpdate)
+        repository.save(it)
```

Comment:
can we delegate the repository interactions to the service layer?

Reply:
yes -- could it be done separately from this PR?

Reply:
yep. karen looking at this this afternoon 👍